stablehlo/reference/InterpreterOps.cpp

```diff
@@ -42,6 +42,9 @@ namespace stablehlo {
 namespace interpreter {
 namespace {
 
+// A strictly increasing counter to uniquely serialize instrumented probe values
+static int64_t serializedProbeId = 0;
```

Comment:
Let's avoid static values. They'll make testing more difficult, among other issues.

How about we store this either in InterpreterConfig or DefaultInterpreterFallback, and pass it to probe op? Probably default fallback is better since it is related to an interpreter dialect op.